app/views/posts/index.html.erb

```diff
@@ -1 +1,33 @@
-<h1>Here is a list of all posts by user 1</h1>
\ No newline at end of file
+<article class="main-article" style="width: 90%; margin: auto">
+  <div style="display: flex; align-items: center; justify-content: space-between;">
+    <%= image_tag @user.photo, :class => "user-avatar" %>
+    <div class="user-details">
+      <p><%= @user.name %></p>
+      <span>Number of posts: <%= @user.post_counter %></span>
+    </div>
+  </div>
+
+  <% @user.recent_posts.each_with_index do |post, index| %>
+    <div>
+      <div class="post">
+        <h4>Post #<%= index + 1 %></h4>
+        <div>
+          <p class="post-title"><%= post.title %></p>
+        </div>
+        <div class="post-footer">
+          <span>Comments: <%= post.comments_counter %></span>
+          <span>Likes: <%= post.likes_counter %></span>
+        </div>
+      </div>
+      <div class="post-comments">
+        <div class="comments">
+          <% post.recent_comments.each do |comment| %>
+            <div class="comment">
+              <span>User <%= comment.user_id %></span> : <span><%= comment.text %></span>
+            </div>
+          <% end %>
```

Comment:
- [ ] [Optional] I noticed that you are repeating the same code with the same style in the `Post#index` and `Post#show` pages. Repeating code goes against the **DRY (Don't Repeat Yourself)** principle. To address this, I recommend using [Rails partials](https://guides.rubyonrails.org/layouts_and_rendering.html#using-partials) to reduce code duplication. By creating partials, you can encapsulate common code and include it in both views, promoting cleaner and more maintainable code. Please consider updating your code to improve its structure and adhere to best practices. Keep up the good work! 🚀🧹